bika/lims/catalog/indexers/analysisrequest.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of SENAITE.CORE
+#
+# Copyright 2018 by it's authors.
+# Some rights reserved. See LICENSE.rst, CONTRIBUTORS.rst.
+
+from bika.lims import api
+from bika.lims.interfaces import IAnalysisRequest
+from bika.lims.workflow import getCurrentState
+from plone.indexer import indexer
+
+
+@indexer(IAnalysisRequest)
+def assigned_state(instance):
+    """Returns `assigned` or `unassigned` depending on the state of the
+    analyses the analysisrequest contains. Return `unassigned` if the Analysis
+    Request does not contain any analysis or if has at least one in `unassigned`
+    state. Otherwise, returns `assigned`"""
+    analyses = instance.getAnalyses()
+    if not analyses:
+        return 'unassigned'
+
+    for analysis in analyses:
+        analysis = api.get_object(analysis)
+        state = getCurrentState(analysis, 'worksheetanalysis_review_state')
```

Comment:
please use `api.get_workflow_status_of` instead, then you don't have to ensure the object in line 25 first: https://github.com/senaite/senaite.core/blob/master/bika/lims/api.py#L740